sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -13,171 +13,178 @@
 #include <CL/sycl/detail/common.hpp>
 #include <CL/sycl/detail/os_util.hpp>
 #include <CL/sycl/detail/pi.h>
+#include <cassert>
+#include <string>
+
+// Function to load the shared library
+// Implementation is OS dependent.
+void *loadOsLibrary(const std::string &library);
+
+// Function to get Address of a symbol defined in the shared
+// library, Implementation is OS dependent.
+void *getOsLibraryFuncAddress(void *Library, const std::string &FunctionName);
```

Comment:
Wondering if the LLVM Support class [DynamicLibrary](http://llvm.org/doxygen/classllvm_1_1sys_1_1DynamicLibrary.html) could be used here - though there might be unintended interaction with the libraries loaded and globally stored by a LLVM runtime compiler.

Reply:
You are right, too much baggage and unwanted behavior.